Source/WebCore/dom/Observable.cpp

```diff
@@ -25,33 +25,53 @@
 #include "config.h"
 #include "Observable.h"
 
+#include "AbortSignal.h"
 #include "CallbackResult.h"
 #include "Exception.h"
 #include "ExceptionCode.h"
+#include "InternalObserver.h"
+#include "JSSubscriptionObserverCallback.h"
 #include "SubscribeOptions.h"
 #include "Subscriber.h"
 #include "SubscriberCallback.h"
 #include "SubscriptionObserver.h"
-#include "SubscriptionObserverCallback.h"
 
 namespace WebCore {
 
-ExceptionOr<Ref<Observable>> Observable::create(Ref<SubscriberCallback> callback)
+ExceptionOr<Ref<Observable>> Observable::create(RefPtr<SubscriberCallback> callback)
 {
     return adoptRef(*new Observable(callback));
 }
 
 void Observable::subscribe(ScriptExecutionContext& context, std::optional<ObserverUnion> observer, SubscribeOptions options)
+{
+    if (observer.has_value()) {
+        WTF::switchOn(
+            observer.value(),
+            [&](RefPtr<JSSubscriptionObserverCallback>& next) {
+                subscribeInternal(context, adoptRef(*new InternalObserverFromScript(context, next)), options);
```

Comment:
These aren't really FromScript, since they're being created here in C++. Does it make sense to have InternalObserver be pure virtual if there's only one class which inherits from it?

Reply:
It looks a bit silly in this PR but subsequent to this, each operator (e.g. take, takeUntil, filter, map) will each implement their own InternalObserver specialisation.